### ISSUE TITLE
ambiguous `version` helper for rhel_sysvinit service provider

### ISSUE THREAD

**Issue by stevenolen** [chef-cookbooks/httpd#56]
https://github.com/chef-cookbooks/httpd/blob/master/libraries/httpd_service_rhel_sysvinit.rb#L13
and
https://github.com/chef-cookbooks/httpd/blob/master/libraries/httpd_service_rhel_sysvinit.rb#L23

refer to a `version` helper(?) that collides with `chef-sugar` when they are both in a node's run_list. Modifying both of these lines to use `new_resource.version` resolves the issue in my environment.

```
[2015-12-15T15:53:08-05:00] DEBUG: Re-raising exception: ArgumentError - shib_oauth2_bridge[default] (ectg-ucnext::_bridge line 48) had an error: ArgumentError: httpd_service_rhel_sysvinit[shib-oauth2-bridge-default] (/var/chef/cache/cookbooks/shib-oauth2-bridge/libraries/provider_shib_oauth2_bridge.rb line 19) had an error: ArgumentError: wrong number of arguments (0 for 1)
/opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-sugar-3.2.0/lib/chef/sugar/constraints.rb:137:in `version'
  /var/chef/cache/cookbooks/httpd/libraries/httpd_service_rhel_sysvinit.rb:13:in `block (2 levels) in <class:HttpdServiceRhelSysvinit>'
  /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.5.1/lib/chef/resource_builder.rb:77:in `instance_eval'
  /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.5.1/lib/chef/resource_builder.rb:77:in `build'
  /var/chef/cache/cookbooks/compat_resource/files/lib/chef_compat/copied_from_chef/chef/dsl/declare_resource.rb:109:in `build_resource'
  /var/chef/cache/cookbooks/compat_resource/files/lib/chef_compat/copied_from_chef/chef/dsl/declare_resource.rb:67:in `declare_resource'
  /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.5.1/lib/chef/dsl/resources.rb:15:in `template'
  /var/chef/cache/cookbooks/httpd/libraries/httpd_service_rhel_sysvinit.rb:12:in `block in <class:HttpdServiceRhelSysvinit>'
  (eval):2:in `block in action_start'
  /var/chef/cache/cookbooks/compat_resource/files/lib/chef_compat/copied_from_chef/chef/provider.rb:118:in `instance_eval'
  /var/chef/cache/cookbooks/compat_resource/files/lib/chef_compat/copied_from_chef/chef/provider.rb:118:in `compile_and_converge_action'
  (eval):2:in `action_start'
  /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.5.1/lib/chef/provider.rb:144:in `run_action'
  /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.5.1/lib/chef/resource.rb:585:in `run_action'
  /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.5.1/lib/chef/runner.rb:49:in `run_action'
  /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.5.1/lib/chef/runner.rb:81:in `block (2 levels) in converge'
  /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.5.1/lib/chef/runner.rb:81:in `each'
  /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.5.1/lib/chef/runner.rb:81:in `block in converge'
  /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.5.1/lib/chef/resource_collection/resource_list.rb:83:in `block in execute_each_resource'
  /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.5.1/lib/chef/resource_collection/stepable_iterator.rb:116:in `call'
  /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.5.1/lib/chef/resource_collection/stepable_iterator.rb:116:in `call_iterator_block'
  /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.5.1/lib/chef/resource_collection/stepable_iterator.rb:85:in `step'
  /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.5.1/lib/chef/resource_collection/stepable_iterator.rb:104:in `iterate'
  /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.5.1/lib/chef/resource_collection/stepable_iterator.rb:55:in `each_with_index'
  /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.5.1/lib/chef/resource_collection/resource_list.rb:81:in `execute_each_resource'
...
```

**Comment by tpetchel** [comment]
@stevenolen: I'm glad you filed this issue. I encountered the same and thought I was going crazy :)

I verified your fix and submitted #60.

**Comment by stevenolen** [comment]
thanks!
